### ISSUE TITLE
Problems upgrading from Pods 2.7.31 to 2.9.13

### ISSUE THREAD

**Issue by techprogrammer** [pods-framework/pods#7033]
### Description

Upgrading from 2.7.31 pods to present release 2.9.13. Wordpress 6.2.

Repaired all Pods.
Select Edit Pod on vendors pods (It repairs it again)
With this pod, it created several extra fields for most.

Example:

Label    Name            Type
Active  active             Yes/No
Active  active_135             Yes/No
Active  active_147             Yes/No
Active  active_159             Yes/No

### Version

2.7.31

### Testing Instructions

Update from 2.7.31 to 2.9.13.

Repair all Pods
Select Edit Pod on vendors pod

### Screenshots / Screencast

![Untitled](https://user-images.githubusercontent.com/5952458/226255425-2d269e03-552d-4c9c-b5d9-b4dea4a34973.png)

![field types](https://user-images.githubusercontent.com/5952458/226415803-f7f4c5f8-f324-4608-9a8f-ee3ec4af4743.PNG)

**Comment by JoryHogeveen** [comment]
@sc0ttkclark any pointers here?

**Comment by sc0ttkclark** [comment]
Not sure yet, I'll have to go through and look at the tool that does this. I can only think of one case where this could happen when it cannot find the fields at the time it runs this and it attempts to add them to the first group. Maybe there's a cache situation there

**Comment by JoryHogeveen** [comment]
Alright, thanks for the update!

**Comment by techprogrammer** [comment]
> This person has experienced this issue and a few others with the upgrade/fixer tool that auto runs after updating Pods and going to the edit pod screen for the first time. I haven't been able to reproduce the exact scenario before but I've added more and more checks and handling for cases that could cause it at this point.

scott, let me know what I can send/do to help replicate it.  I've emailed you the export migrate package in past email.  Appreciate the help!

**Comment by techprogrammer** [comment]
This issue still happens in Pods 3.0.0b1

**Comment by sc0ttkclark** [comment]
Digging into this again

**Comment by techprogrammer** [comment]
2.9.15 has same issue.

Test script
Upgrade to 2.9.5
Website works ok

Select edit Pods
Edit Categories pod, it does a bunch of repairs
Edit Deals pod, it does a bunch of repairs
Edit Vendors pod, it does a bunch of repairs
Website gets SQL errors loading

Repair all pods

Website gets SQL errors loading

**Comment by techprogrammer** [comment]
2.9.16 has similar issues as 2.9.15 with Vendors getting lots of extra fields made.

After the auto repairs, the website loads, no more sql error at least

**Comment by techprogrammer** [comment]
2.9.18 has similar issues as 2.9.16 with Vendors getting lots of extra fields made.

**Comment by sc0ttkclark** [comment]
I spent time today digging into what was going on here. I found out that the site prior to 2.8+ already had duplicate field configurations saved in the DB due to a previous bug in 2.X

Prior to the tool, these configurations would go unnoticed and only some times cause unexplained issues where a field may have old saved settings show up. Now the tool exposes those and attempts to save them as unique fields for further manual review.

Looks like this is good to close now after doing the digging.
